BlueprintUI/Sources/Layout/ConstrainedAspectRatio.swift

```diff
@@ -66,13 +78,14 @@ public struct ConstrainedAspectRatio: Element {
                     } else if constrainedHeight > availableHeight {
                         return aspectRatio.size(forHeight: constrainedHeight)
                     }
+                    return contentSize
                 } else if constraint.width.constrainedValue == nil {
                     return aspectRatio.size(forWidth: constrainedWidth)
                 } else if constraint.height.constrainedValue == nil {
                     return aspectRatio.size(forHeight: constrainedHeight)
                 } else if constrainedWidth < availableWidth {
                     return aspectRatio.size(forHeight: constrainedHeight)
-                } else if constrainedHeight < availableHeight {
+                } else {
                     return aspectRatio.size(forWidth: constrainedWidth)
                 }
```

Comment:
When the constraint matched the available size we were falling all the way out of this switch and returning `contentSize` at the bottom of the method. I removed that fallback and made every switch case return something appropriate.